dpcpp/CMakeLists.txt

```diff
@@ -82,8 +82,10 @@ configure_file(preconditioner/jacobi_common.hpp.in preconditioner/jacobi_common.
 ginkgo_compile_features(ginkgo_dpcpp)
 target_compile_definitions(ginkgo_dpcpp PRIVATE GKO_COMPILING_DPCPP _ONEDPL_COMPILE_KERNEL=0)
 
+set(GINKGO_DPCPP_FLAGS "-fsycl")
```

Comment:
Does this flag also work with the `dpcpp` compiler?

Reply:
Also, we could try rewriting this this form:
https://github.com/oneapi-src/oneAPI-samples/pull/1721/files#diff-cbe919c8af7559493969dbc2c78bf849e49a594bd82fc018982f948266dbee24R1

Note: it requires a higher CMake version, also probably a high oneAPi version. I see IntelSYCLConfig provided in 2023.1. Before it was IntelDPCPP. See:
2023.0 instructions: https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2023-0/use-cmake-with-the-compiler.html
2023.1 instructions: https://www.intel.com/content/www/us/en/docs/dpcpp-cpp-compiler/developer-guide-reference/2023-1/use-cmake-with-the-compiler.html

Reply:
yes, dpcpp has the flag, too.
I went through IntelDpcpp.config. there's no `add_sycl_to_target` in this module.